Automated export to nextcloud folder does not work (on android 14)

### Checklist

- [X] I have used the search function for [**OPEN**](https://github.com/AntennaPod/AntennaPod/issues) issues to see if someone else has already submitted the same bug report.
- [X] I have **also** used the search function for [**CLOSED**](https://github.com/AntennaPod/AntennaPod/issues?q=is%3Aissue+is%3Aclosed) issues to see if the problem is already solved and just waiting to be released.
- [X] I will describe the problem with as much detail as possible.
- [X] If the bug only to occurs with a certain podcast, I will include the URL of that podcast.

### App version

3.4.0

### Where did you get the app from

Google Play

### Android version

14

### Device model

Pixel 8

### First occurred

did not work since ever (with this version) for me

### Steps to reproduce

1. Open settings, import/export -> automated export 
2. Select folder in nextcloud media storage
3. Grant access to folder
4. It does make and upload backup without issues 
5. When the automatic export should happen the next time I get a push, that the export did not work!


### Expected behaviour

Automated export can access folder and does export the database

Path should be:
/storage/emulated/0/Android/media/com.nextcloud.client/nextcloud/name@domain.next-cloud.org/AntennaPod

### Current behaviour

Automated export does not work.

My guess is, that the path to folder has an issue - see log:
Caused by: java.io.FileNotFoundException: Temp folder could not be created: /null/nextcloud/tmp/name@domain.next-cloud.org

but funny enough it does work when I activate the function, select the folder and grant access!(It does then create the file, store it it the selected folder and upload it to my nextcloud.

Is there maybe an issue with the access control or the stored path?

### Logs

## Environment
Android version: 14
OS version: 5.15.131-android14-11-gd99d9fe08e0c-ab11209359
AntennaPod version: 3.4.0
Model: Pixel 8
Device: shiba
Product: shiba

No crash report recorded

--- a/app/src/main/java/de/danoeh/antennapod/ui/screen/preferences/ImportExportPreferencesFragment.java
+++ b/app/src/main/java/de/danoeh/antennapod/ui/screen/preferences/ImportExportPreferencesFragment.java
@@ -400,7 +400,8 @@ public class ImportExportPreferencesFragment extends AnimatedPreferenceFragment 
         @Override
         public Intent createIntent(@NonNull final Context context, @Nullable final Uri input) {
             return super.createIntent(context, input)
-                    .addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION | Intent.FLAG_GRANT_WRITE_URI_PERMISSION);
+                    .addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION | Intent.FLAG_GRANT_WRITE_URI_PERMISSION
+                            | Intent.FLAG_GRANT_PERSISTABLE_URI_PERMISSION);
         }
     }
 

